### ISSUE TITLE
Annotation/descriptors to get binding info from slice types

### ISSUE THREAD

**Issue by sbose78** [redhat-developer/service-binding-operator#361]
https://github.com/redhat-developer/service-binding-operator/issues/352
 
feature request from @navidsh :

"What do you guys think about using annotation/descriptors to get binding info from slice types?

Example:
```yaml
status:
  listeners:
    - addresses:
        - host: my-cluster-kafka-bootstrap.coffeeshop.svc
          port: 9092
      type: plain
...
```

**Comment by arthurdm** [comment]
It would be good to solve that for the spec in [this section](https://github.com/application-stacks/service-binding-specification#pointer-to-binding-data) as well.  Right now it assumes the statusDescriptors are all top level status items.

**Comment by navidsh** [comment]
@sbose78 Yea, I was able to use the Go template to get the binding info I wanted.
BTW, my GitHub id is `navidsh` and not `navidshaikh` 😜

**Comment by sbose78** [comment]
haha thanks for the confirmation. @navidshaikh @navidsh , greetings both of you :P

**Comment by sbose78** [comment]
> It would be good to solve that for the spec in this section as well. Right now it assumes the statusDescriptors are all top level status items.

+1 .

We should allow unlimited levels of nesting in every block in the CR.

**Comment by pedjak** [comment]
Fixed by #964
